src/content/docs/ai-monitoring/configure-ai-monitoring.mdx

```diff
@@ -0,0 +1,187 @@
+---
+title: 'Configure AI monitoring'
+metaDescription: 'You can apply certain configurations to your APM agents to change how your AI data appears in New Relic.'
+freshnessValidatedDate: never
+---
+
+Once you install AI monitoring, you can configure the default behavior of the agent or update your app to collect different kinds of data. 
+
+## Configure the agent [#configure-agents]
+
+Update default agent behavior for AI monitoring at these agent configuration docs: 
+
+* [Go](/docs/apm/agents/go-agent/configuration/go-agent-configuration/#ai-monitoring)
+* [Node.js](/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration/#ai-monitoring)
+* [Python](/docs/apm/agents/python-agent/configuration/python-agent-configuration/#ai-monitoring)
+* [Ruby](/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#ai-monitoring)
+
+## Enable token count API [#enable-token]
+
+If you've disabled `ai_monitoring.record_content.enabled`, you may still want to know the number of tokens your app uses in an interaction. If you haven't disabled content from being recorded by the agent, you don't need to invoke the token count API. 
+
+Update your app code with the below code snippets: 
+
+<CollapserGroup>
+    <Collapser
+        id="go-token-api"
+        title="Go"
+    >
+
+        INSERT_TEXT_HERE
+
+    </Collapser>
+    <Collapser
+        id="nodejs-token-api"
+        title="Node.js"
+    >
+
+    ```js
+    // register callback to set token counts 
+      newrelic.setLlmTokenCount(callback)
+
+      /**
+      * Sample callback to calculate token count. Must be a synchronous method.
+      * This callback is optional, and is designed for customers that cannot send
+      * the content in the Llm events but still want token counts in the case
+      * they are not present.
+      * @param {string} model name of model
+      * @param {string} content content of call
+      * @returns {int} token count for content, customers must use relevant libraries to calculate this
+      */
+      function callback(model, content) {
+        // put your business logic to calculate token counts
+        return 10
+      }
+    ```
+
+    </Collapser>
+    <Collapser
+        id="python-token-api"
+        title="Python"
+    >
+        INSERT_TEXT_HERE
+    </Collapser>
+    <Collapser
+        id="ruby-token-api"
```

Comment:
Ruby Example:

```ruby
require 'tiktoken_ruby'

def token_count_callback(model, content)
    if model.includes? "gpt"
        enc = Tiktoken.encoding_for_model(model)
        return enc.encode(content).length
    end

    # for models that are not supported in the example
    return 0
end

NewRelic::Agent.set_llm_token_count_callback(token_count_callback)
```

Reply:
I did not leave the method comments on the ruby code as I did not quickly find a canonical way to comment it. One of the docs I found recommends no comments; fun!